src/components/Todos.tsx

```diff
@@ -0,0 +1,271 @@
+import React, { useEffect, useState } from "react";
+import "bootstrap/dist/css/bootstrap.css";
+import "../stylesheets/Todos.css";
+import { useNavigate } from "react-router-dom";
+import { ArrowUp, ArrowDown } from "lucide-react";
+import useFetch from "../hooks/useFetch";
+import { useQuery, useMutation } from "react-query";
+import axios from "axios";
+import { Todo } from "../types/Todo";
+
+export default function Todos() {
+  // let { data, isLoading, error,refetch } = useFetch("http://localhost:8000/todos");
+  const [todolist, setTodolist] = useState<Todo[]>([]);
+  const [search, setSearch] = useState<string>("");
+  const [status, setStatus] = useState("all");
+  const [page, setPage] = useState<number>(1);
+  const [limit, setLimit] = useState<number>(2);
+  const [totalpages, setTotalPages] = useState(2);
+  const [sortBy, setSortBy] = useState<string>("name");
+  const [sortOrder, setSortOrder] = useState<string>("asc");
+  const { data, error, isLoading, refetch } = useQuery({
+    queryKey: ["todos", page, limit, search, sortBy, sortOrder],
+    queryFn: async () => {
+      const res = await axios.get(
+        // `http://localhost:8000/todos?_page=${page}&_per_page=${limit}&`
+        `http://localhost:8000/todos?_page=${page}&_limit=${limit}&title_like=${search}&_sort=${sortBy}&_order=${sortOrder}&isCompleted_like=${status}`
```

Comment:
Axios provide a params key where you can send query params

Reply:
Done